inc/customizer.php

```diff
@@ -61,3 +107,173 @@ function twentynineteen_customize_preview_js() {
+				.image-filters-enabled .site-header.featured-image .site-featured-image:before,
+				.image-filters-enabled .site-header.featured-image .site-featured-image:after,
+				.image-filters-enabled .entry .post-thumbnail:before,
+				.image-filters-enabled .entry .post-thumbnail:after,
+				.entry-content .wp-block-button .wp-block-button__link,
+				.button, button, input[type="button"], input[type="reset"], input[type="submit"],
+				.main-navigation .sub-menu,
+				.sticky-post,
+				.entry-content > *[class^="wp-block-"].has-primary-background-color,
+				.entry-content > *[class^="wp-block-"] .has-primary-background-color,
+				.entry-content > *[class^="wp-block-"].is-style-solid-color .has-primary-background-color {
+					background-color: %1$s;
+				}
+
+				/* Set Color for:
+				 * - all links
+				 * - main navigation links
+				 * - Post navigation links
+				 * - Post entry meta hover
+				 * - Post entry header more-link hover
+				 * - main navigation svg
+				 * - comment navigation
+				 * - Comment edit link hover
+				 * - Site Footer Link hover
+				 */
+				a,
+				a:visited,
+				.main-navigation ul.main-menu > li > a,
+				.post-navigation .post-title,
+				.entry .entry-meta a:hover,
+				.entry .entry-footer a:hover,
+				.entry .entry-content .more-link:hover,
+				.main-navigation .main-menu > li > a + svg,
+				.comment-navigation .nav-previous a:hover,
+				.comment-navigation .nav-next a:hover,
+				.comment .comment-metadata .comment-edit-link:hover,
+				.site-footer a:hover,
+				.entry-content .wp-block-button.is-style-outline .wp-block-button__link:not(.has-text-color),
+				.entry-content .wp-block-button.is-style-outline .wp-block-button__link:focus:not(.has-text-color),
+				.entry-content .wp-block-button.is-style-outline .wp-block-button__link:active:not(.has-text-color),
+				.entry-content > *[class^="wp-block-"] .has-primary-color,
+				.entry-content > *[class^="wp-block-"].is-style-solid-color .has-primary-color {
+					color: %1$s;
+				}
+
+				/* Set left border color for:
+				 * wp block quote
+				 */
+				.entry-content .wp-block-quote:not(.is-large), .entry-content .wp-block-quote:not(.is-style-large) {
+					border-left-color: %1$s;
+				}
+
+				/* Set border color for:
+				 * :focus
+				 */
+				input[type="text"]:focus,
+				input[type="email"]:focus,
+				input[type="url"]:focus,
+				input[type="password"]:focus,
+				input[type="search"]:focus,
+				input[type="number"]:focus,
+				input[type="tel"]:focus,
+				input[type="range"]:focus,
+				input[type="date"]:focus,
+				input[type="month"]:focus,
+				input[type="week"]:focus,
+				input[type="time"]:focus,
+				input[type="datetime"]:focus,
+				input[type="datetime-local"]:focus,
+				input[type="color"]:focus,
+				textarea:focus {
+					border-color: %1$s
+				}
+
+				.gallery-item > div > a:focus {
+					box-shadow: 0 0 0 2px %1$s;
+				}
+	';
+
+	wp_add_inline_style( 'twentynineteen-style', sprintf( $css, $primary_color ) );
```

Comment:
Similar to Twenty Seventeen and other themes that use instant custom color previews, we need to expose a data attribute with a the color in the customize preview here. This allows us to avoid duplicating the CSS rules in JS. Instead of `wp_add_inline_style`, this should be:

```
	$customize_preview_data_color = '';
	if ( is_customize_preview() ) {
		$customize_preview_data_color = 'data-color="' . $primary_color . '"';
	}
?>
	<style type="text/css" id="twentynineteen-colors" <?php echo $customize_preview_data_color; ?>>
		<?php echo $css; ?>
	</style>
<?php
```